2025-08-29 14:26:48,210 : INFO : root : Loading config.
2025-08-29 14:26:48,214 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-08-29 14:26:48,215 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\White\Documents\Top_Gun_Pinball\config\config.yaml
2025-08-29 14:26:48,215 : INFO : ConfigProcessor : Loading config from file C:\Users\White\mpf\Lib\site-packages\mpf\mpfconfig.yaml.
2025-08-29 14:26:48,228 : INFO : ConfigProcessor : Loading config: C:\Users\White\mpf\Lib\site-packages\mpf\mpfconfig.yaml
2025-08-29 14:26:48,229 : INFO : ConfigProcessor : Loading config from file C:\Users\White\Documents\Top_Gun_Pinball\config\config.yaml.
2025-08-29 14:26:48,230 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config.yaml
2025-08-29 14:26:48,230 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_ball_devices.yaml
2025-08-29 14:26:48,231 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_coils.yaml
2025-08-29 14:26:48,233 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\config\config_switches.yaml
2025-08-29 14:26:48,235 : INFO : ConfigProcessor : Config file cache created: C:\Users\White\AppData\Local\Temp\cfa6ecba65704a931c9dd214c44986c8.mpf_cache
2025-08-29 14:26:48,343 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-08-29 14:26:48,344 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\91cc4d1c993c58dc354e34974b9cfeac.mpf_cache
2025-08-29 14:26:48,345 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-08-29 14:26:48,345 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\76792f1d686f89709887dba7bd69667a.mpf_cache
2025-08-29 14:26:48,346 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\142bbae4707335cc8e3efaa4364c4893.mpf_cache
2025-08-29 14:26:48,347 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev9
2025-08-29 14:26:48,347 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2025-08-29-14-26-48-mpf-TopGunPinball.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'full_logfile_path': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-08-29-14-26-48-mpf-TopGunPinball.log'}
2025-08-29 14:26:48,347 : INFO : Machine : MPF path: C:\Users\White\mpf\Lib\site-packages\mpf
2025-08-29 14:26:48,347 : INFO : Machine : Machine path: C:\Users\White\Documents\Top_Gun_Pinball
2025-08-29 14:26:48,347 : INFO : Machine : Platform: win32
2025-08-29 14:26:48,347 : INFO : Machine : Python executable location: C:\Users\White\mpf\Scripts\python.exe
2025-08-29 14:26:48,347 : INFO : Machine : Python version: 3.12.7 (64-bit)
2025-08-29 14:26:48,350 : INFO : Machine : Initializing MPF...
2025-08-29 14:26:48,451 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,481 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-08-29 14:26:48,481 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,481 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,482 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.7', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-11-10.0.26100-SP0', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '11', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.26100', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-08-29-14-26-48-mpf-TopGunPinball.log', 'prev_value': None, 'change': True}
2025-08-29 14:26:48,603 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-08-29 14:26:48,631 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-08-29 14:26:48,632 : INFO : SwitchController : Found active switch: (1003, <Platform.Virtual>) <switch.s_trough_1>
2025-08-29 14:26:48,632 : INFO : SwitchController : Found active switch: (1004, <Platform.Virtual>) <switch.s_trough_2>
2025-08-29 14:26:48,632 : INFO : SwitchController : Found active switch: (1005, <Platform.Virtual>) <switch.s_trough_3>
2025-08-29 14:26:48,632 : INFO : SwitchController : Found active switch: (1006, <Platform.Virtual>) <switch.s_trough_4>
2025-08-29 14:26:48,632 : INFO : SwitchController : Found active switch: (1007, <Platform.Virtual>) <switch.s_trough_5>
2025-08-29 14:26:48,633 : INFO : SwitchController : Found active switch: (1008, <Platform.Virtual>) <switch.s_trough_6>
2025-08-29 14:26:48,634 : INFO : EventManager : Event: ======'bcp_connection_attempt'====== Args={'name': 'local_display', 'host': 'localhost', 'port': 5050}
2025-08-29 14:26:48,635 : INFO : EventManager : Event: ======'bcp_connection_attempt'====== Args={'name': 'server', 'host': '0.0.0.0', 'port': 5050}
2025-08-29 14:26:48,635 : INFO : BCPClientSocket : Connecting BCP to 'local_display' at localhost:5050...
2025-08-29 14:26:48,637 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...Error('type')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...Error('type')>)>
Traceback (most recent call last):
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\bcp\bcp.py", line 78, in _setup_bcp_connections
    client = Util.string_to_class(settings['type'])(self.machine, name, self.machine.bcp)
                                  ~~~~~~~~^^^^^^^^
KeyError: 'type'
2025-08-29 14:26:48,657 : INFO : Machine : Starting the main run loop with active modes: []
2025-08-29 14:26:48,657 : WARNING : Machine : Attract mode is not active, game will not be playable. Please check your attract mode configuration.
2025-08-29 14:26:48,658 : INFO : BCPClientSocket : Connected BCP to 'local_display' localhost:5050
2025-08-29 14:26:48,659 : INFO : Machine : Shutting down...
2025-08-29 14:26:48,659 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-08-29 14:26:49,674 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\machine.py", line 808, in _run_loop
    raise self._exception['exception']
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\bcp\bcp.py", line 78, in _setup_bcp_connections
    client = Util.string_to_class(settings['type'])(self.machine, name, self.machine.bcp)
                                  ~~~~~~~~^^^^^^^^
KeyError: 'type'
2025-08-29 14:26:49,677 : INFO : root : MPF run loop ended.
